src/invocation/RegistrationKey.ts

```diff
@@ -0,0 +1,42 @@
+export class RegistrationKey {
+    private readonly userRegistrationId: string;
+    private handler: Function;
+    private encoder: Function;
```

Comment:
Can we rename all encoder and decoder naming to give the meaning of

* `registration Encoder Function` ( something like `registerEncodeFunc`) 
* `registration Decoder Function`
* `deregistration Encoder Function`
* `deregistration Decoder Function`

I can see encode decode keywords everywhere, it's hard to follow.

Reply:
did that.